File: interpolacao_permutacao_python/Px_assembly.py
from permutacao import Permutation


class PAssembly:

    def __init__(self, coefficients, vector):
        self.vector = vector
        self.vector_modified = []
        self.coefficients = coefficients
        self.coefficient = None
        self.perm = None
        self.vector_mod()

    def vector_mod(self):
        """
        :rtype : object
        """
        for objects in self.vector:
                self.vector_modified.append(-objects)

    def assembly(self):
        n = self.coefficients.__len__()
        self.coefficient = n * [0]
        self.perm = Permutation(self.vector_modified)
        """
            invertendo o sinal da biblioteca px0

        """
        self.coefficient[0] = self.coefficients[0]
        print("Px inicial = "),
        print self.coefficient
        print("----------------------------")
        for x in range(1, n):
            if not self.coefficients[x]:
                continue
            else:
                self.assembly_mount(self.coefficients[x], x)
            print("Px[%d] = " % (self.coefficient.__len__())),
            print self.coefficient
            print("----------------------------")
        else:
            return self.coefficient

    def assembly_mount(self, coefficient, n):

        """
        :param self:
        :param coefficient:
        :param n:
        """
        vector_permutation = []
        coefficient_parcial = n * [0]
        for k in range(1, n + 1):
            vector_permutation = self.perm.permutation(n, k, '*')
            soma = 0
            for x in range(vector_permutation.__len__()):
                soma += vector_permutation[x]
            self.coefficient[n - k] += (coefficient * soma)
            coefficient_parcial[n - k] = coefficient * soma
            print("n=%d||k=%d||vectorperm[%d] = " % (n, k, vector_permutation.__len__())),
            print vector_permutation
            print ("coefficient_parcial[%d] = " % (coefficient_parcial.__len__())),
            print coefficient_parcial
        else:
            self.coefficient[n] += coefficient

File: IE_KNIN_JAVAFX/src/sample/IPKnin/Pxassembly.java
package sample.IPKnin;

/**
 * Created by lisboa on 03/04/2015.
 */

public class Pxassembly {

    private double[] vector;
    private double[] coefficients;

    private double[] vector_mod;
    private double[] degrees;
    private Permutation perm;

    public Pxassembly(double[] coefficients, double[] vector) {
        this.coefficients = coefficients;
        this.vector = vector;

        vector_mod = new double[this.vector.length];
        Vector_invert();
        this.perm = new Permutation(this.vector_mod);
    }

    private void Vector_invert() {
        for (int x = 0; x < this.vector.length; x++) {
            vector_mod[x] = -vector[x];
        }
    }

    public double[] assembly() {
        int n = this.coefficients.length;
        this.degrees = new double[n];
        this.degrees[0] = this.coefficients[0];

       /* System.out.print("Px inicial = [");
        for (double objects : this.degrees)
            System.out.print(objects + ",");
        System.out.print("]\n----------------------------\n");
*/
        for (int x = 1; x < n; x++) {
            if (this.coefficients[x] != 0)
                this.assembly_mount(this.coefficients[x], x);
            /*System.out.print("Px["+(this.degrees.length)+"] = [");
            for (double objects : this.degrees) {
                System.out.print(objects+",");
            }
            System.out.print("]\n----------------------------\n");
            */
        }
        return this.degrees;
    }

    private void assembly_mount(double coefficient, int n) {

        double[] vector_permutation;
        //double[] coefficient_parcial = new double[n];

        double soma;

        for(int  k = 1 ; k < (n+1);k++) {
            vector_permutation = this.perm.permutation(n, k, '*');
            soma = 0;

            for (int x = 0; x < vector_permutation.length; x++) {
                soma += vector_permutation[x];
            }

            this.degrees[n - k] += (coefficient * soma);
            //coefficient_parcial[n - k] = coefficient * soma;
            /*
            System.out.print("n="+n+"||k="+k+"||vectorperm["+vector_permutation.length+"]=[");
            for (double objects : vector_permutation) {
                System.out.print(objects+",");
            }
            System.out.print("]\n");

            System.out.print("coefficient_parcial["+coefficient_parcial.length+"] = [");
            for (double objects : coefficient_parcial) {
                System.out.print(objects+",");
            }
            System.out.print("]\n");
            */
        }
        this.degrees[n] += coefficient;
    }
}
